apps/site/assets/ts/schedule/components/LineDiagram.tsx

```diff
@@ -97,120 +128,196 @@ const LineDiagram = ({
+                    : ""
+                }
+                ${
+                  isBranchStart(index)
+                    ? "m-schedule-diagram__stop--branch-start"
+                    : ""
+                }
+              `}
             >
-              <svg>
-                <g transform="translate(0 -4)">
-                  <circle cx="5" cy="5" r="4" strokeWidth="2" />
-                </g>
-              </svg>
-            </div>
-            <div className="m-schedule-line-diagram__card">
-              <div className="m-schedule-line-diagram__card-left">
-                <div className="m-schedule-line-diagram__stop-name">
-                  {maybeAlert(stopAlerts)}
-                  <a href={`/stops/${routeStop.id}`}>
-                    <h4>{routeStop.name}</h4>
-                  </a>
+              <div
+                style={{ color: `#${route.color}` }}
+                className="m-schedule-diagram__lines"
+              >
+                <div className="m-schedule-diagram__line">
+                  {!isOnSecondaryBranch(lineDiagramStop) && (
+                    <svg>
+                      <circle
+                        r="4"
+                        cx="5"
+                        cy={isBranchStart(index) ? "2" : "26"}
+                      />
+                    </svg>
+                  )}
                 </div>
-                <div className="m-schedule-line-diagram__connections">
-                  {filteredConnections(routeStop.connections).map(
-                    (connectingRoute: Route) => (
-                      <TooltipWrapper
-                        key={connectingRoute.id}
-                        href={`/schedules/${connectingRoute.id}/line`}
-                        tooltipText={connectionName(connectingRoute)}
-                        tooltipOptions={{
-                          placement: "bottom",
-                          animation: "false"
-                        }}
-                      >
-                        {connectingRoute.type === 3 ? (
-                          <span
-                            key={connectingRoute.id}
-                            className={`c-icon__bus-pill--small m-schedule-line-diagram__connection ${busBackgroundClass(
-                              connectingRoute
-                            )}`}
-                          >
-                            {connectingRoute.name}
-                          </span>
-                        ) : (
-                          <span
+
+                {isOnSecondaryBranch(lineDiagramStop) && (
+                  <div className="m-schedule-diagram__line m-schedule-diagram__line--branch">
+                    <svg>
+                      {isMergeStop(lineDiagramStop) && (
+                        <path
+                          d={`
+                            M-10,${isBranchStart(index) ? "-30" : "100"}
+                            h 15 
+                            v ${isBranchStart(index) ? "36" : "-76"}
```

Comment:
Will the merge-line-drawing work if there are multiple branches? (e.g. Green Line)

Reply:
The logic in the latest commit should work for any number of extra branches! 💃